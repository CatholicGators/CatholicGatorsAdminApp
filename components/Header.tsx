import React from 'react';

import { withStyles, createStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import AccountCircle from '@material-ui/icons/AccountCircle';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import Button from '@material-ui/core/Button';
import { User } from 'lib/withAuth';

const styles = createStyles({
    root: {
        flexGrow: 1,
    },
    grow: {
        flexGrow: 1,
    },
    menuButton: {
        marginLeft: -12,
        marginRight: 20,
    },
});

interface HeaderProps {
    user: User;
    classes: any;
}

class Header extends React.Component {
    state = {
        anchorEl: null,
    };

    constructor(public props: HeaderProps) {
        super(props);
    }

    handleMenu(event) {
        this.setState({ anchorEl: event.currentTarget });
    };

    handleClose() {
        this.setState({ anchorEl: null });
    };

    render() {
        const { user, classes } = this.props;
        const { anchorEl } = this.state;

        return (
            <div className={classes.root}>
                <AppBar position="static">
                    <Toolbar>
                        <IconButton className={classes.menuButton} color="inherit" aria-label="Menu">
                            <MenuIcon />
                        </IconButton>
                        <Typography variant="h6" color="inherit" className={classes.grow}>
                            Catholic Gators Admin
                        </Typography>
                        {user ? (
                            <div>
                                <IconButton
                                    aria-owns={anchorEl ? 'menu-appbar' : undefined}
                                    aria-haspopup="true"
                                    onClick={this.handleMenu.bind(this)}
                                    color="inherit"
                                >
                                    <AccountCircle />
                                </IconButton>
                                <Menu
                                    id="menu-appbar"
                                    anchorEl={anchorEl}
                                    anchorOrigin={{
                                        vertical: 'top',
                                        horizontal: 'right',
                                    }}
                                    transformOrigin={{
                                        vertical: 'top',
                                        horizontal: 'right',
                                    }}
                                    open={anchorEl}
                                    onClose={this.handleClose.bind(this)}
                                >
                                    <MenuItem onClick={this.handleClose.bind(this)}>Profile</MenuItem>
                                    <MenuItem onClick={this.handleClose.bind(this)}>My account</MenuItem>
                                </Menu>
                            </div>
                        ) : (
                            <Button color="inherit">Login</Button>
                        )}
                    </Toolbar>
                </AppBar>
            </div>
        );
    }
}

export default withStyles(styles)(Header);
