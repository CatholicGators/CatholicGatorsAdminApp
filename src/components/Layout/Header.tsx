import React from 'react';

import { withStyles, createStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Avatar from '@material-ui/core/Avatar';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import Button from '@material-ui/core/Button';

import withAuthentication from '../Auth/withAuthentication';

const styles = createStyles({
    root: {
        flexGrow: 1,
    },
    grow: {
        flexGrow: 1,
    },
    menuButton: {
        marginLeft: -12,
        marginRight: 20,
    },
});

class Header extends React.Component {
    state = {
        anchorEl: null,
    };

    constructor(public props: any) {
        super(props);
    }

    handleMenu(event) {
        this.setState({ anchorEl: event.currentTarget });
    }

    handleClose() {
        this.setState({ anchorEl: null });
    }

    handleLogin() {
        this.props.googleSignIn();
    }

    handleLogout() {
        this.props.signOut();
        this.handleClose();
    }

    render() {
        const { user, classes, children } = this.props;
        const { anchorEl } = this.state;

        return (
            <div className={classes.root}>
                <AppBar position="static">
                    <Toolbar>
                        <IconButton className={classes.menuButton} color="inherit" aria-label="Menu">
                            <MenuIcon />
                        </IconButton>
                        <Typography variant="h6" color="inherit" className={classes.grow}>
                            Catholic Gators Admin
                        </Typography>
                        {user ? (
                            <div>
                                <IconButton
                                    aria-owns={anchorEl ? 'menu-appbar' : undefined}
                                    aria-haspopup="true"
                                    onClick={this.handleMenu.bind(this)}
                                    color="inherit"
                                >
                                    <Avatar src={user.photoURL} />
                                </IconButton>
                                <Menu
                                    id="menu-appbar"
                                    anchorEl={anchorEl}
                                    anchorOrigin={{
                                        vertical: 'top',
                                        horizontal: 'right',
                                    }}
                                    transformOrigin={{
                                        vertical: 'top',
                                        horizontal: 'right',
                                    }}
                                    open={!!anchorEl}
                                    onClose={this.handleClose.bind(this)}
                                >
                                    <MenuItem onClick={this.handleLogout.bind(this)}>Logout</MenuItem>
                                </Menu>
                            </div>
                        ) : (
                            <Button color="inherit" onClick={this.handleLogin.bind(this)}>Login</Button>
                        )}
                    </Toolbar>
                </AppBar>
                {children}
            </div>
        );
    }
}

export default withAuthentication(withStyles(styles)(Header));