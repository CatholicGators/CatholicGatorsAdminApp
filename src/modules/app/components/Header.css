.root, .grow {
    flex-grow: 1;
}

.menuButton {
    margin-left: -12;
    margin-right: 20;
}