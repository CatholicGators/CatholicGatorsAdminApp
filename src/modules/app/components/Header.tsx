import React from 'react';
import { connect } from 'react-redux';
import { createStyles, withStyles } from '@material-ui/core';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Avatar from '@material-ui/core/Avatar';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import Button from '@material-ui/core/Button';
// import CircularProgress from '@material-ui/core/CircularProgress';

import { googleSignIn, signOut } from '../../../redux/actions/auth/authActions';

const styles = createStyles({
    root: {
        flexGrow: 1
    },
    grow: {
        flexGrow: 1
    },
    menuButton: {
        marginLeft: -12,
        marginRight: 20
    }
});

type Props = {
    user: any,
    signOut: () => any,
    googleSignIn: () => any
}

export class Header extends React.Component<Props> {
    state = {
        anchorEl: null,
    };

    constructor(public props) {
        super(props);
    }

    handleMenu(event) {
        this.setState({ anchorEl: event.currentTarget });
    }

    handleClose() {
        this.setState({ anchorEl: null });
    }

    handleLogin() {
        this.props.googleSignIn();
    }

    handleLogout() {
        this.props.signOut();
        this.handleClose();
    }

    profile(user) {
        const { anchorEl } = this.state;

        switch(user) {
            case undefined:
                return <div>Loading...</div>; // TODO: Replace this with CircularProgress

            case null:
                return <Button id="login-btn" color="inherit" onClick={this.handleLogin.bind(this)}>Login</Button>

            default:
                return <div>
                    <IconButton
                        id="avatar-btn"
                        aria-owns={anchorEl ? 'menu' : undefined}
                        aria-haspopup="true"
                        onClick={this.handleMenu.bind(this)}
                        color="inherit"
                    >
                        <Avatar src={user.photoURL} />
                    </IconButton>
                    <Menu
                        id="menu"
                        anchorEl={anchorEl}
                        anchorOrigin={{
                            vertical: 'top',
                            horizontal: 'right',
                        }}
                        transformOrigin={{
                            vertical: 'top',
                            horizontal: 'right',
                        }}
                        open={!!anchorEl}
                        onClose={this.handleClose.bind(this)}
                    >
                        <MenuItem id="logout" onClick={this.handleLogout.bind(this)}>Logout</MenuItem>
                    </Menu>
                </div>
        }
    }

    render() {
        const { user, classes } = this.props;
        // const { anchorEl } = this.state;

        return (
            <div className={classes ? classes.root : null}>
                <AppBar position="static">
                    <Toolbar>
                        <IconButton className={classes ? classes.menuButton : null} color="inherit" aria-label="Menu">
                            <MenuIcon />
                        </IconButton>
                        <Typography variant="h6" color="inherit" className={classes ? classes.grow : null}>
                            Catholic Gators Admin
                        </Typography>
                        {this.profile(user)}
                    </Toolbar>
                </AppBar>
            </div>
        );
    }
}

const mapStateToProps = state => ({
    user: state.auth.user
});

const mapDispatchToProps = dispatch => ({
    googleSignIn: () => dispatch(googleSignIn()),
    signOut: () => dispatch(signOut())
});

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(Header));
