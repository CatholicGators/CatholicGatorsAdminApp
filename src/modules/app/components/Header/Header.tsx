import React from 'react';
import { connect } from 'react-redux';
import { withRouter, NavLink } from "react-router-dom";

import { createStyles, withStyles, Theme } from '@material-ui/core';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import VpnKey from '@material-ui/icons/VpnKey';
import FormatAlignLeft from '@material-ui/icons/FormatAlignLeft';
import Inbox from '@material-ui/icons/Inbox';

import { googleSignIn, signOut } from '../../../../redux/actions/auth/authActions';
import MobileDrawer from '../MobileDrawer/MobileDrawer';
import ToolbarAvatar from '../ToolbarAvatar/ToolbarAvatar';

const styles = (theme: Theme) => createStyles({
    root: {
        flexGrow: 1
    },
    grow: {
        flexGrow: 1
    },
    menuButton: {
        marginLeft: -12,
        marginRight: 20,
        [theme.breakpoints.up('md')]: {
            display: 'none'
        }
    },
    toolbar: {
        minHeight: '64px'
    },
    desktopMenuItems: {
        display: 'flex',
        [theme.breakpoints.down('sm')]: {
            display: 'none'
        }
    },
    navLinkDesktop: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        paddingLeft: 20,
        paddingRight: 20,
        color: 'inherit',
        textDecoration: 'none',
        '&:visited': {
            color: 'inherit'
        },
        '&:hover': {
            color: theme.palette.secondary.main,
            transition: 'color',
            transitionDuration: '200ms',
        }
    },
    navLinkTextDesktop: {
        color: 'inherit'
    },
    desktopSelected: {
        color: theme.palette.secondary.main
    }
});

type Props = {
    user: any,
    signOut: () => any,
    googleSignIn: () => any
}

type State = {
    drawerOpen: boolean
}

export type MenuLink = {
    text: string,
    href: string,
    icon: any,
    needsAuthorization: boolean
}
export const menuLinks: Array<MenuLink> = [
    {
        text: 'Contact Form',
        href: '/',
        icon: FormatAlignLeft,
        needsAuthorization: false
    },
    {
        text: 'My Contacts',
        href: '/my-contacts',
        icon: Inbox,
        needsAuthorization: false
    },
    {
        text: 'Admin',
        href: '/admin',
        icon: VpnKey,
        needsAuthorization: true
    }
];

export class Header extends React.Component<Props, State> {
    state = {
        drawerOpen: false
    };

    constructor(public props) {
        super(props);
    }

    toggleDrawer(isOpen: boolean){
      this.setState({
        drawerOpen: isOpen,
      });
    };

    render() {
        const { classes, user } = this.props;

        return (
            <div className={classes ? classes.root : null}>
                <AppBar position="static">
                    <Toolbar className={classes ? classes.toolbar : null}>
                        <IconButton
                            id='menu-btn'
                            className={classes ? classes.menuButton : null}
                            color="inherit"
                            aria-label="Menu"
                            onClick={() => this.toggleDrawer(true)}
                        >
                            <MenuIcon/>
                        </IconButton>
                        <Typography variant="h6" color="inherit" className={classes ? classes.grow : null}>
                            Catholic Gators Admin
                        </Typography>
                        <div className={classes ? classes.desktopMenuItems : null}>
                            { user ?
                                menuLinks
                                    .filter(link => !link.needsAuthorization || (user.isAdmin && user.isApproved))
                                    .map(link => 
                                            <NavLink
                                                exact
                                                key={link.text}
                                                to={link.href}
                                                className={classes ? classes.navLinkDesktop : null}
                                                activeClassName={classes ? classes.desktopSelected : null}
                                            >
                                                <Typography
                                                    className={classes ? classes.navLinkTextDesktop : null}
                                                >
                                                    {link.text}
                                                </Typography>
                                            </NavLink>
                                    )
                                : null
                            }
                            <ToolbarAvatar
                                user={user}
                                login={() => this.props.googleSignIn()}
                                logout={() => this.props.signOut()}
                            />
                        </div>
                    </Toolbar>
                </AppBar>
                <MobileDrawer
                    isOpen={this.state.drawerOpen}
                    user={user}
                    closeDrawer={() => this.toggleDrawer(false)}
                    menuLinks={menuLinks}
                    logout={() => this.props.signOut() }
                    login={() => this.props.googleSignIn() }
                    selectedPath={this.props.location.pathname}
                />
            </div>
        );
    }
}

const mapStateToProps = state => ({
    user: state.auth.user
});

const mapDispatchToProps = dispatch => ({
    googleSignIn: () => dispatch(googleSignIn()),
    signOut: () => dispatch(signOut())
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(Header)));
