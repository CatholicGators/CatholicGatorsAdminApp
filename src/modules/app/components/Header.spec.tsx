import React from 'react'
import { shallow } from 'enzyme';

import Header from './Header';

describe('Header.tsx', () => {
    let props, wrapper;

    beforeEach(() => {
        props = {
            user: null,
            signOut: jest.fn(),
            googleSignIn: jest.fn()
        };
        wrapper = shallow(<Header {...props} />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    it('should show the login button when user is null', () => {
        expect(wrapper.exists('#login-btn'));
    });
});