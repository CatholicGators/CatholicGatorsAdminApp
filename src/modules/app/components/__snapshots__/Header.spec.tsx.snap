// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    googleSignIn={[MockFunction]}
    signOut={[MockFunction]}
    user={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(AppBar)
        position="static"
      >
        <WithStyles(Toolbar)>
          <WithStyles(IconButton)
            aria-label="Menu"
            className={null}
            color="inherit"
          >
            <pure(MenuIcon) />
          </WithStyles(IconButton)>
          <WithStyles(Typography)
            className={null}
            color="inherit"
            variant="h6"
          >
            Catholic Gators Admin
          </WithStyles(Typography)>
          <div>
            <WithStyles(IconButton)
              aria-haspopup="true"
              color="inherit"
              id="avatar-btn"
              onClick={[Function]}
            >
              <WithStyles(Avatar) />
            </WithStyles(IconButton)>
            <WithStyles(Menu)
              anchorEl={null}
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              id="menu"
              onClose={[Function]}
              open={false}
              transformOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
            >
              <WithStyles(MenuItem)
                id="logout"
                onClick={[Function]}
              >
                Logout
              </WithStyles(MenuItem)>
            </WithStyles(Menu)>
          </div>
        </WithStyles(Toolbar)>
      </WithStyles(AppBar)>,
      "className": null,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Toolbar)>
          <WithStyles(IconButton)
            aria-label="Menu"
            className={null}
            color="inherit"
          >
            <pure(MenuIcon) />
          </WithStyles(IconButton)>
          <WithStyles(Typography)
            className={null}
            color="inherit"
            variant="h6"
          >
            Catholic Gators Admin
          </WithStyles(Typography)>
          <div>
            <WithStyles(IconButton)
              aria-haspopup="true"
              color="inherit"
              id="avatar-btn"
              onClick={[Function]}
            >
              <WithStyles(Avatar) />
            </WithStyles(IconButton)>
            <WithStyles(Menu)
              anchorEl={null}
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              id="menu"
              onClose={[Function]}
              open={false}
              transformOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
            >
              <WithStyles(MenuItem)
                id="logout"
                onClick={[Function]}
              >
                Logout
              </WithStyles(MenuItem)>
            </WithStyles(Menu)>
          </div>
        </WithStyles(Toolbar)>,
        "position": "static",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(IconButton)
              aria-label="Menu"
              className={null}
              color="inherit"
            >
              <pure(MenuIcon) />
            </WithStyles(IconButton)>,
            <WithStyles(Typography)
              className={null}
              color="inherit"
              variant="h6"
            >
              Catholic Gators Admin
            </WithStyles(Typography)>,
            <div>
              <WithStyles(IconButton)
                aria-haspopup="true"
                color="inherit"
                id="avatar-btn"
                onClick={[Function]}
              >
                <WithStyles(Avatar) />
              </WithStyles(IconButton)>
              <WithStyles(Menu)
                anchorEl={null}
                anchorOrigin={
                  Object {
                    "horizontal": "right",
                    "vertical": "top",
                  }
                }
                id="menu"
                onClose={[Function]}
                open={false}
                transformOrigin={
                  Object {
                    "horizontal": "right",
                    "vertical": "top",
                  }
                }
              >
                <WithStyles(MenuItem)
                  id="logout"
                  onClick={[Function]}
                >
                  Logout
                </WithStyles(MenuItem)>
              </WithStyles(Menu)>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Menu",
              "children": <pure(MenuIcon) />,
              "className": null,
              "color": "inherit",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Catholic Gators Admin",
              "className": null,
              "color": "inherit",
              "variant": "h6",
            },
            "ref": null,
            "rendered": "Catholic Gators Admin",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <WithStyles(IconButton)
                  aria-haspopup="true"
                  color="inherit"
                  id="avatar-btn"
                  onClick={[Function]}
                >
                  <WithStyles(Avatar) />
                </WithStyles(IconButton)>,
                <WithStyles(Menu)
                  anchorEl={null}
                  anchorOrigin={
                    Object {
                      "horizontal": "right",
                      "vertical": "top",
                    }
                  }
                  id="menu"
                  onClose={[Function]}
                  open={false}
                  transformOrigin={
                    Object {
                      "horizontal": "right",
                      "vertical": "top",
                    }
                  }
                >
                  <WithStyles(MenuItem)
                    id="logout"
                    onClick={[Function]}
                  >
                    Logout
                  </WithStyles(MenuItem)>
                </WithStyles(Menu)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "aria-haspopup": "true",
                  "aria-owns": undefined,
                  "children": <WithStyles(Avatar) />,
                  "color": "inherit",
                  "id": "avatar-btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "src": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "anchorEl": null,
                  "anchorOrigin": Object {
                    "horizontal": "right",
                    "vertical": "top",
                  },
                  "children": <WithStyles(MenuItem)
                    id="logout"
                    onClick={[Function]}
                  >
                    Logout
                  </WithStyles(MenuItem)>,
                  "id": "menu",
                  "onClose": [Function],
                  "open": false,
                  "transformOrigin": Object {
                    "horizontal": "right",
                    "vertical": "top",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Logout",
                    "id": "logout",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Logout",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(AppBar)
          position="static"
        >
          <WithStyles(Toolbar)>
            <WithStyles(IconButton)
              aria-label="Menu"
              className={null}
              color="inherit"
            >
              <pure(MenuIcon) />
            </WithStyles(IconButton)>
            <WithStyles(Typography)
              className={null}
              color="inherit"
              variant="h6"
            >
              Catholic Gators Admin
            </WithStyles(Typography)>
            <div>
              <WithStyles(IconButton)
                aria-haspopup="true"
                color="inherit"
                id="avatar-btn"
                onClick={[Function]}
              >
                <WithStyles(Avatar) />
              </WithStyles(IconButton)>
              <WithStyles(Menu)
                anchorEl={null}
                anchorOrigin={
                  Object {
                    "horizontal": "right",
                    "vertical": "top",
                  }
                }
                id="menu"
                onClose={[Function]}
                open={false}
                transformOrigin={
                  Object {
                    "horizontal": "right",
                    "vertical": "top",
                  }
                }
              >
                <WithStyles(MenuItem)
                  id="logout"
                  onClick={[Function]}
                >
                  Logout
                </WithStyles(MenuItem)>
              </WithStyles(Menu)>
            </div>
          </WithStyles(Toolbar)>
        </WithStyles(AppBar)>,
        "className": null,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Toolbar)>
            <WithStyles(IconButton)
              aria-label="Menu"
              className={null}
              color="inherit"
            >
              <pure(MenuIcon) />
            </WithStyles(IconButton)>
            <WithStyles(Typography)
              className={null}
              color="inherit"
              variant="h6"
            >
              Catholic Gators Admin
            </WithStyles(Typography)>
            <div>
              <WithStyles(IconButton)
                aria-haspopup="true"
                color="inherit"
                id="avatar-btn"
                onClick={[Function]}
              >
                <WithStyles(Avatar) />
              </WithStyles(IconButton)>
              <WithStyles(Menu)
                anchorEl={null}
                anchorOrigin={
                  Object {
                    "horizontal": "right",
                    "vertical": "top",
                  }
                }
                id="menu"
                onClose={[Function]}
                open={false}
                transformOrigin={
                  Object {
                    "horizontal": "right",
                    "vertical": "top",
                  }
                }
              >
                <WithStyles(MenuItem)
                  id="logout"
                  onClick={[Function]}
                >
                  Logout
                </WithStyles(MenuItem)>
              </WithStyles(Menu)>
            </div>
          </WithStyles(Toolbar)>,
          "position": "static",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(IconButton)
                aria-label="Menu"
                className={null}
                color="inherit"
              >
                <pure(MenuIcon) />
              </WithStyles(IconButton)>,
              <WithStyles(Typography)
                className={null}
                color="inherit"
                variant="h6"
              >
                Catholic Gators Admin
              </WithStyles(Typography)>,
              <div>
                <WithStyles(IconButton)
                  aria-haspopup="true"
                  color="inherit"
                  id="avatar-btn"
                  onClick={[Function]}
                >
                  <WithStyles(Avatar) />
                </WithStyles(IconButton)>
                <WithStyles(Menu)
                  anchorEl={null}
                  anchorOrigin={
                    Object {
                      "horizontal": "right",
                      "vertical": "top",
                    }
                  }
                  id="menu"
                  onClose={[Function]}
                  open={false}
                  transformOrigin={
                    Object {
                      "horizontal": "right",
                      "vertical": "top",
                    }
                  }
                >
                  <WithStyles(MenuItem)
                    id="logout"
                    onClick={[Function]}
                  >
                    Logout
                  </WithStyles(MenuItem)>
                </WithStyles(Menu)>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Menu",
                "children": <pure(MenuIcon) />,
                "className": null,
                "color": "inherit",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Catholic Gators Admin",
                "className": null,
                "color": "inherit",
                "variant": "h6",
              },
              "ref": null,
              "rendered": "Catholic Gators Admin",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <WithStyles(IconButton)
                    aria-haspopup="true"
                    color="inherit"
                    id="avatar-btn"
                    onClick={[Function]}
                  >
                    <WithStyles(Avatar) />
                  </WithStyles(IconButton)>,
                  <WithStyles(Menu)
                    anchorEl={null}
                    anchorOrigin={
                      Object {
                        "horizontal": "right",
                        "vertical": "top",
                      }
                    }
                    id="menu"
                    onClose={[Function]}
                    open={false}
                    transformOrigin={
                      Object {
                        "horizontal": "right",
                        "vertical": "top",
                      }
                    }
                  >
                    <WithStyles(MenuItem)
                      id="logout"
                      onClick={[Function]}
                    >
                      Logout
                    </WithStyles(MenuItem)>
                  </WithStyles(Menu)>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "aria-haspopup": "true",
                    "aria-owns": undefined,
                    "children": <WithStyles(Avatar) />,
                    "color": "inherit",
                    "id": "avatar-btn",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "src": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "anchorEl": null,
                    "anchorOrigin": Object {
                      "horizontal": "right",
                      "vertical": "top",
                    },
                    "children": <WithStyles(MenuItem)
                      id="logout"
                      onClick={[Function]}
                    >
                      Logout
                    </WithStyles(MenuItem)>,
                    "id": "menu",
                    "onClose": [Function],
                    "open": false,
                    "transformOrigin": Object {
                      "horizontal": "right",
                      "vertical": "top",
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Logout",
                      "id": "logout",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Logout",
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
